host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncServerStateOperationHandler.java

```diff
@@ -117,8 +123,12 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                     final String opName = entry.getValue() == SyncServerResultAction.RESTART_REQUIRED ?
                             ServerProcessStateHandler.REQUIRE_RESTART_OPERATION : ServerProcessStateHandler.REQUIRE_RELOAD_OPERATION;
                     final OperationStepHandler handler = context.getResourceRegistration().getOperationHandler(serverAddress, opName);
-                    final ModelNode op = Util.createEmptyOperation(opName, serverAddress);
-                    context.addStep(op, handler, OperationContext.Stage.MODEL);
+                    if (handler != null) {
+                        final ModelNode op = Util.createEmptyOperation(opName, serverAddress);
+                        context.addStep(op, handler, OperationContext.Stage.MODEL);
+                    } else {
+                        throw HostControllerLogger.ROOT_LOGGER.failedToSyncServerStatus(opName, serverAddress.toString());
+                    }
```

Comment:
In our June 1 chat we talked about logging a WARN here, not throwing an exception. I think this is fine though; if this is null it's a ??? situation and we should probably fail.

Reply:
👍